### PR TITLE
Don't use a stale extent for a non-global base layer

### LINKED ISSUE #2815: Exception in ImageryLayer._createTileImagerySkeletons
This is in Cesium 1.10 and perhaps many prior versions.

Run this [Sandcastle example](http://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=Imagery%20Layers%20Manipulation.html&label=Showcases).  Uncheck all imagery sources, then check `TileMapServer Image`.

image

### PR DESCRIPTION
**Pull request by kring**
Broken logic could leave a rectangle in `_createTileImagerySkeletons` with stale data.  Mostly this was harmless-ish, but if the stale data happened to lie outside the valid bounds of the imagery layer's tiling scheme, it could cause the exception seen in #2815.

Fixes #2815

### CODE REVIEW

File: Specs/Scene/ImageryLayerSpec.js [hunk 1]
```diff
@@ -267,6 +267,80 @@ defineSuite([
...
+                expect(tiles[1].data.imagery[0].textureCoordinateRectangle.x).toBe(0.0);
+                expect(tiles[1].data.imagery[0].textureCoordinateRectangle.w).toBe(1.0);
+                expect(tiles[1].data.imagery[0].textureCoordinateRectangle.z).toBe(1.0);
+                expect(tiles[1].data.imagery[1].textureCoordinateRectangle.x).toBe(0.0);
+                expect(tiles[1].data.imagery[1].textureCoordinateRectangle.y).toBe(0.0);
+                expect(tiles[1].data.imagery[1].textureCoordinateRectangle.z).toBe(1.0);
+            });
+        });
+
+        /*it('does not get confused when base layer imagery overlaps in one direction but not the other', function() {
```
Comment (pjcozzi) [inline Specs/Scene/ImageryLayerSpec.js]: What do you want to do with this commented out test?

Comment (kring) [inline Specs/Scene/ImageryLayerSpec.js]: Oops, sorry, meant to delete that.

**Comment by kring** [comment]
Commented-out code is gone.

**Comment by pjcozzi** [comment]
Tests and Sandcastle example are good, thanks.

**Pull request merged**